### ISSUE TITLE
Soft WDT Reset?

### ISSUE THREAD

**Issue by beicnet** [bblanchon/ArduinoJson#626]
Hi @bblanchon,

For some reason, after request to Weather Underground API (every 10 mins) I got "Soft WDT Reset", I tought it's because I'm using the older version of your library v5.6.7, now I updated it from the Git to v5.11.2, but same issue continues.

```
Connected!

{
  "response": {
  "version":"0.1",
  "termsofService":"http://www.wunderground.com/weather/api/d/terms.html",
  "features": {
  "conditions": 1
  }
	}
  ,	"current_observation": {
		"image": {
		"url":"http://icons.wxug.com/graphics/wu2/logo_130x80.png",
		"title":"Weather Underground",
		"link":"http://www.wunderground.com"
		},
		"display_location": {
		"full":"Becej, Serbia",
		"city":"Becej",
		"state":"06",
		"state_name":"Serbia",
		"country":"RB",
		"country_iso3166":"RS",
		"zip":"00000",
		"magic":"8",
		"wmo":"13168",
		"latitude":"45.61999893",
		"longitude":"20.04999924",
		"elevation":"75.9"
		},
		"observation_location": {
		"full":"Becej, Gornji Salas 60, Bečej, ",
		"city":"Becej, Gornji Salas 60, Bečej",
		"state":"",
		"country":"RS",
		"country_iso3166":"RS",
		"latitude":"45.634960",
		"longitude":"20.013065",
		"elevation":"255 ft"
		},
		"estimated": {
		},
		"station_id":"IBEEJ2",
		"observation_time":"Last Updated on November 29, 9:45 PM CET",
		"observation_time_rfc822":"Wed, 29 Nov 2017 21:45:51 +0100",
		"observation_epoch":"1511988351",
		"local_time_rfc822":"Wed, 29 Nov 2017 21:46:06 +0100",
		"local_epoch":"1511988366",
		"local_tz_short":"CET",
		"local_tz_long":"Europe/Belgrade",
		"local_tz_offset":"+0100",
		"weather":"Partly Cloudy",
		"temperature_string":"40.3 F (4.6 C)",
		"temp_f":40.3,
		"temp_c":4.6,
		"relative_humidity":"78%",
		"wind_string":"From the SSE at 8.3 MPH Gusting to 12.3 MPH",
		"wind_dir":"SSE",
		"wind_degrees":148,
		"wind_mph":8.3,
		"wind_gust_mph":"12.3",
		"wind_kph":13.4,
		"wind_gust_kph":"19.8",
		"pressure_mb":"1009",
		"pressure_in":"29.80",
		"pressure_trend":"0",
		"dewpoint_string":"34 F (1 C)",
		"dewpoint_f":34,
		"dewpoint_c":1,
		"heat_index_string":"NA",
		"heat_index_f":"NA",
		"heat_index_c":"NA",
		"windchill_string":"35 F (2 C)",
		"windchill_f":"35",
		"windchill_c":"2",
		"feelslike_string":"35 F (2 C)",
		"feelslike_f":"35",
		"feelslike_c":"2",
		"visibility_mi":"6.2",
		"visibility_km":"10.0",
		"solarradiation":"0",
		"UV":"0.0","precip_1hr_string":"0.00 in ( 0 mm)",
		"precip_1hr_in":"0.00",
		"precip_1hr_metric":" 0",
		"precip_today_string":"0.00 in 
Soft WDT reset

ctx: cont 
sp: 3ffef480 end: 3ffef780 offset: 01b0

>>>stack>>>
3ffef630:  3ffe8a08 3ffef690 3ffef700 40204bdc  
3ffef640:  3ffee704 00000883 3fff1c1f 40205b94  
3ffef650:  3ffee610 3ffee610 3ffef700 3ffee724  
3ffef660:  3ffee610 00000000 3ffee724 40205bc5  
3ffef670:  3fffdad0 00000000 3ffee610 40204699  
3ffef680:  3fffdad0 00000000 3ffee748 402028e5  
3ffef690:  3fff05ac 3fff0894 3fff08bc 0000001f  
3ffef6a0:  00000014 fe000050 fe001388 3fff08e4  
3ffef6b0:  0000003f 00000037 3fff0814 0000000f  
3ffef6c0:  00000004 3fff0614 0000000f 00000000  
3ffef6d0:  3fff062c 0000001f 00000011 3fff0654  
3ffef6e0:  0000000f 00000000 00000000 00000000  
3ffef6f0:  000000c8 00000a3c 3ffe8500 00000000  
3ffef700:  00000000 00000000 00000000 3ffee750  
3ffef710:  3fffdad0 0000000a 3ffee724 40204651  
3ffef720:  3ffe85d8 3ffee630 3ffee724 40204651  
3ffef730:  3ffe8458 3ffee750 3ffee724 4020467c  
3ffef740:  3fff008c 40202d58 3ffee724 3ffee750  
3ffef750:  3fffdad0 00000000 3ffee748 40202ab8  
3ffef760:  feefeffe 00000000 3ffee748 40205734  
3ffef770:  feefeffe feefeffe 3ffee760 40100718  
<<<stack<<<
```

Any thoughts about it?

Regards,
Viktor

**Comment by bblanchon** [comment]
Hi Viktor,

Are you reusing the `JsonBuffer`?
If so it's probably filling up until it cannot parse the input anymore.

See:

* [Why does my device crash or reboot?](https://arduinojson.org/faq/why-does-my-device-crash-or-reboot/) (updated a minute ago)
* [How to reuse a `JsonBuffer`?](http://arduinojson.org/faq/how-to-reuse-a-jsonbuffer/)

Regards,
Benoit

**Comment by beicnet** [comment]
Hi @bblanchon,

I just using `DynamicJsonBuffer jsonBuffer;` and calling my function every 10 minutes (made with Millis) from loop.

Also can't use `jsonBuffer.clear();`, because it gives me compiler error (your latest Git version).

The second what I testing now, I commented out `Serial.println(JsonData);` in my sketch, so it will not output the parsed data to the Serial monitor and it seems that the "Soft WDT reset" gone for now.

Is it possible that the Json Serial Output did those WDT resets?!

Also I just found your **ArduinoJson Assistant** and it's just great!!!

Kind regards,
Viktor
